common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -781,8 +819,10 @@ public void onData(String streamId, ByteBuffer buf) throws IOException {
       // to disk as well. This way, we avoid having to buffer the entirety of every blocks in
       // memory, while still providing the necessary guarantee.
       synchronized (partitionInfo) {
-        if (isStaleOrTooLate(appShuffleInfo.shuffles.get(partitionInfo.shuffleId),
-            partitionInfo.shuffleMergeId, partitionInfo.reduceId)) {
+        if (isStale(appShuffleInfo.shuffles.get(partitionInfo.shuffleId),
+            partitionInfo.shuffleMergeId) ||
+            isTooLate(appShuffleInfo.shuffles.get(partitionInfo.shuffleId),
+            partitionInfo.reduceId)) {
```

Comment:
Fetch `appShuffleInfo.shuffles.get(partitionInfo.shuffleId)` and use same instance for both checks.
This pattern is there in `onComplete` and `onFailure` as well.